### PR TITLE
add get filemetadata for list of LFNs. Fix #7204

### LINKED ISSUE #7204: refactor retrieve of filemetadata
We clearly suffer of a cmsweb-independend problem with GET filemetadata since at time Publisher needs to retrieve >100MB in one call which has  a high chance of timing out at 5min. 
The problem starts with the fact that all filemetadata for a given task are retrieved in one operation. For tasks with thousands of jobs and large list of lumis for each file this leads to huge JSON structures.

Solution is not obvious.

This is not really a bug (things are working atm) but as datasets with many lumis become more common, it could be a show stopper.

We may need to find a solution before it becomes critical.

### PR DESCRIPTION
**Pull request by belforte**

### CODE REVIEW

File: src/python/CRABInterface/DataFileMetadata.py [hunk 1]
```diff
@@ -24,17 +25,29 @@ def __init__(self, config):
         self.logger = logging.getLogger("CRABLogger.DataFileMetadata")
         self.FileMetaData = getDBinstance(config, 'FileMetaDataDB', 'FileMetaData')
 
-    def getFiles(self, taskname, filetype, howmany, lfn):
+    def getFiles(self, taskname, filetype, howmany=None, lfnList=[]):
```
[the remark this answers is not in the document]
Comment (belforte) [reply src/python/CRABInterface/DataFileMetadata.py:28]: will change

[the remark this answers is not in the document]
Comment (belforte) [reply src/python/CRABInterface/DataFileMetadata.py]: solved via https://github.com/dmwm/CRABServer/pull/7231/commits/8f78e8cea8f24973d7f0e810e0456c863be44e8d

**Comment by belforte** [comment]
worked in a couple test submissions. I have put new server and publisher on test1 and started Jenkins tests. Let's wait before merging this.
https://github.com/dmwm/CRABServer/issues/7232

**Comment by belforte** [comment]
Jenkins on cmsweb-test2/crab-dev-tw01 succeeded. https://github.com/dmwm/CRABServer/issues/7232
Will test on preprod (maybe long URL's are a problem there ?)

**Review by mapellidario — APPROVED** [review]
This looks good to me. I like the idea of starting by simply checking that the list of files retrieved is identical.

**Comment by belforte** [comment]
thanks @mapellidario and @novicecpp for reviewing. I will test more in preprod before moving to prod.

File: src/python/Publisher/PublisherMaster.py [hunk 2]
```diff
@@ -622,6 +670,15 @@ def startSlave(self, task):
                                  workflow)
                 else:
                     publDescFiles_list = self.getPublDescFiles(workflow, lfn_ready, logger)
+                    publDescFiles_list2 = self.getPublDescFiles2(workflow, lfn_ready, logger)
+                    if not publDescFiles_list == publDescFiles_list:
```
Comment (belforte) [inline src/python/Publisher/PublisherMaster.py]: none of the reviewers spotted this horrible typo !

Comment (belforte) [reply src/python/Publisher/PublisherMaster.py]: fixed in new commit

**Pull request merged**